File: frontend/src/schemas/task.ts
import * as z from "zod";

export type TaskStatus = "TODO" | "IN_PROGRESS" | "DONE";

export const taskSchema = z.object({
  title: z.string().min(1, "Title is required"),
  description: z.string().default(""),
  status: z.enum(["TODO", "IN_PROGRESS", "DONE"]).default("TODO"),
  order_index: z.number().int().default(0),
});

export type TaskFormValues = z.infer<typeof taskSchema>;

export interface Task {
  id: string;
  user_id: string;
  title: string;
  description?: string;
  status: TaskStatus;
  order_index: number;
  created_at: string;
  updated_at: string;
}

export interface TaskReorderItem {
  id: string;
  order_index: number;
}

export interface TaskReorderRequest {
  items: TaskReorderItem[];
}

File: backend/schemas/task.py
from datetime import datetime
from enum import Enum
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TaskStatus(str, Enum):
    TODO = "TODO"
    IN_PROGRESS = "IN_PROGRESS"
    DONE = "DONE"


class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: TaskStatus = TaskStatus.TODO
    order_index: int = 0


class TaskCreate(TaskBase):
    pass


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[TaskStatus] = None
    order_index: Optional[int] = None


class Task(TaskBase):
    id: UUID
    user_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TaskReorderItem(BaseModel):
    id: UUID
    order_index: int


class TaskReorderRequest(BaseModel):
    items: List[TaskReorderItem]
